Second round of updates for the Backstage quickstart

bgood opened the pull request:
The major updates in this round include:
- Code to build and deploy the container images
- Documentation to guide readers on how to update Cloud SQL and deploy the Kubernetes manifests
- Templatization of the Kubernetes manifests and application configs

aburhan commented:
@bgood 

Running terraform apply resulted in the follow issue. 

1. APIs that need to be enabled:
  -   sqladmin.googleapis.com (Cloud SQL Admin API)
  -   compute.googleapis.com (Compute Engine API)
  -   iam.googleapis.com (Identity and Access Management API)
  -   iap.googleapis.com (Cloud Identity-Aware Proxy API)
  -   cloudresourcemanager.googleapis.com (Cloud Resource Manager API)
  -   autoscaling.googleapis.com (Autoscaling API)
  -   containerfilesystem.googleapis.com (Container Filesystem API)

2. Deprecation warning (FYI)
    "Warning: Deprecated Resource
    │
    │    with google_iap_brand.backstageIapBrand,
    │    on iap.tf line 15, in resource "google_iap_brand" "backstageIapBrand":
    │    15: resource "google_iap_brand" "backstageIapBrand" {
    │
    │ after July 2025, the `google_iap_brand` Terraform resource will no longer function as intended due to the deprecation of  the IAP OAuth Admin API"
  
    this says it must be setup in the console...https://github.com/hashicorp/terraform-provider-google/issues/21378

3.  Readme step 6 has a typo
    ```sh
    gcloud container clusters get-credentials backstage-qs --region us-central1 --dns-endpoint   # this should be--dns-endpoint
    ```

bgood commented:
@aburhan Thank you!

1. The list of APIs that get enabled are in `variables.tf` and they are enabled in `project.tf`. IAM was missing but I've added it to the list.  The others were there but because they are enabled async, there are race conditions in the Terraform, I haven't tracked down yet.  Figured we could take care of that in a different PR.
2. Good catch, let's discuss.
3. Good catch, fixed.

bgood commented on reference-architectures/backstage/backstage-quickstart/README.md:
```diff
@@ -120,6 +158,84 @@ Management API are enabled.
+
+    This will take approximately 10 minutes to build and push the image.
+
+7.  Configure Cloud SQL postgres user for password authentication.
```
Backstage itself does use IAM auth the postgres instance, but we need to use the postgres user to grant the CREATEDB privilege, which requires setting a password for the postgres user.

bgood commented on reference-architectures/backstage/backstage-quickstart/README.md:
```diff
@@ -86,15 +86,6 @@ In this section you prepare your project for deployment.
-    sed -i "s/YOUR_STATE_BUCKET/${BACKSTAGE_QS_STATE_BUCKET}/g" ${BACKSTAGE_QS_BASE_DIR}/backend.tf
-    sed -i "s/YOUR_PROJECT_ID/${PROJECT_ID}/g" ${BACKSTAGE_QS_BASE_DIR}/backstage-qs.auto.tfvars
-    sed -i "s/YOUR_IAP_USER_DOMAIN/${IAP_USER_DOMAIN}/g" ${BACKSTAGE_QS_BASE_DIR}/backstage-qs.auto.tfvars
-    sed -i "s/YOUR_IAP_SUPPORT_EMAIL/${IAP_SUPPORT_EMAIL}/g" ${BACKSTAGE_QS_BASE_DIR}/backstage-qs.auto.tfvars
```
Done
